backend/dal/Repositories/PersonRepository.cs

```diff
@@ -103,6 +103,14 @@ public PimsPerson Add(PimsPerson person, bool userOverride)
             }
 
             this.Context.PimsPeople.Add(person);
+
+            // Ensure we don't add existing address from linked organization again
+            var mailing = person.GetMailingAddress();
+            if (person.UseOrganizationAddress == true && mailing != null)
```

Comment:
I see person.UseOrganizationAddress being used here to not create the linkage again. Am I correct in thinking that the actual logic to associate the organization address when this flag is true is on the frontend?

Reply:
Right. When checkbox is checked, the frontend gets the organization details (incl address) and sets it as mailing address. This logic here is to avoid an error I saw on creating a person with existing mailing address. Note that Context.Add(person) marks the person (and all subentities within it) as Added so it was trying to re-insert an existing address. The DB complains due to triggers enforcing no duplicates so this logic basically ignores that address (and only that one)

Reply:
The other use case where someone adds a net new mailing address (not the one linked from org) would not apply to this logic and would be inserted in the DB as expected

Reply:
Also, to respond to your specific question about linkage here. The linkage IS created here. What is not re-created is the address row it points to:

```
person (add to db) -> person-address (add to db) -> address (existing already, do not add)
```

Hopefully that makes it clear